soh/src/code/z_play.c

```diff
@@ -428,12 +445,10 @@ void Play_Init(GameState* thisx) {
         }
     }
 
-    // Invalid entrance, so immediately exit the game to opening title
     if (gSaveContext.entranceIndex == -1) {
         gSaveContext.entranceIndex = 0;
         play->state.running = false;
         SET_NEXT_GAMESTATE(&play->state, Opening_Init, OpeningContext);
-        GameInteractor_ExecuteOnExitGame(gSaveContext.fileNum);
```

Comment:
This doesn't look like an intentional change; I think you need to merge in the latest changes in develop.